src/actions/MastodonAddKernelsAction.C

```diff
@@ -0,0 +1,135 @@
+  {
+    for (std::size_t i = 0; i < _problem->mesh().dimension(); i++)
+    {
+      InputParameters params = _factory.getValidParams("InertialForce");
+      params.set<NonlinearVariableName>("variable") = disp_variables[i];
+      params.set<std::vector<VariableName>>("velocity") = {vel_variables[i]};
+      params.set<std::vector<VariableName>>("acceleration") = {accel_variables[i]};
+      params.set<Real>("beta") = getParam<Real>("beta");
+      params.set<Real>("gamma") = getParam<Real>("gamma");
+      _problem->addKernel("InertialForce", "inertia" + comp[i], params);
+    }
+  }
+
+  // Adding AuxKernels
+  if (auxkernels && _current_task == "add_kernel")
+  {
+    // velocity and acceleration AuxKernels
+    const std::vector<std::string> vel_auxkernel = {"vel_x", "vel_y", "vel_z"};
+    const std::vector<std::string> accel_auxkernel = {"accel_x", "accel_y", "accel_z"};
+    const Real beta = getParam<Real>("beta");
+    const Real gamma = getParam<Real>("gamma");
+    for (std::size_t j = 0; j < _problem->mesh().dimension(); j++)
+    {
+      // Velocity AuxKernels: These Auxkernels are added as MooseObjectActions
+      // due to conflicts with DynamicTensorMechanicsAction when added using
+      // addAuxKernel method. TODO: Investigate this conflict. Talk to Ben Spencer.
+      InputParameters action_params = _action_factory.getValidParams("AddKernelAction");
+      // Create the action
+      action_params.set<std::string>("type") = "NewmarkVelAux";
+      std::shared_ptr<MooseObjectAction> vel_action = std::static_pointer_cast<MooseObjectAction>(
+          _action_factory.create("AddKernelAction", vel_auxkernel[j], action_params));
+      vel_action->appendTask("add_aux_kernel");
+      // Assigning input parameters
+      InputParameters & vel_kernel_params = vel_action->getObjectParams();
+      vel_kernel_params.set<AuxVariableName>("variable") = vel_variables[j];
+      vel_kernel_params.set<std::vector<VariableName>>("acceleration") = {accel_variables[j]};
+      vel_kernel_params.set<Real>("gamma") = gamma;
+      vel_kernel_params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+      // Add the action to the warehouse
+      _awh.addActionBlock(vel_action);
+
+      // Acceleration AuxKernels: Added using the addAuxKernel method
+      InputParameters accel_kernel_params = _factory.getValidParams("NewmarkAccelAux");
+      accel_kernel_params.set<AuxVariableName>("variable") = accel_variables[j];
+      accel_kernel_params.set<std::vector<VariableName>>("displacement") = {disp_variables[j]};
+      accel_kernel_params.set<std::vector<VariableName>>("velocity") = {vel_variables[j]};
+      accel_kernel_params.set<Real>("beta") = beta;
+      accel_kernel_params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_END;
+      _problem->addAuxKernel("NewmarkAccelAux", accel_auxkernel[j], accel_kernel_params);
+    }
+
+    // stress and strain AuxKernels: Added using the addAuxKernel method.
```

Comment:
Can we have a separate boolean variable for the strain/stress output? Storing all 12 auxvariables for all 3D scenarios might be expensive. The velocity and acceleration auxkernels are mandatory to run the dynamic simulation whereas it will be good if we can turn off these output variables for large problems.

Reply:
@sveerara Sure, that sounds like a good idea.